packages/react-components/source/react/library/tooltips/TooltipHoverArea.js

```diff
@@ -42,23 +53,120 @@ const TooltipHoverArea = ({
   tooltip,
   onClick,
   style,
-  enabled,
+  disabled,
+  position,
+  textAlign,
+  ...popperOptions
 }) => {
+  // Tooltip references
+  const [arrowElement, setArrowReference] = useState(null);
+  const [referenceElement, setReferenceElement] = useState(null);
+  const [popperElement, setPopperElement] = useState(null);
+  const { current: tooltipId } = useRef(
+    `tooltip-${Math.floor(Math.random() * 10000)}`,
+  );
+  const popperModifiers = [
+    {
+      name: 'flip',
+      enabled: true,
+    },
+    {
+      name: 'arrow',
+      enabled: popperOptions.arrow,
+      options: {
+        element: arrowElement,
+        padding: 1,
+        offset: [0, 6],
+      },
+    },
+    {
+      name: 'offset',
+      options: {
+        offset: [0, 6],
+      },
+    },
+    {
+      name: 'preventOverflow',
+      options: {
+        rootBoundary: 'document',
+        padding: 0,
+      },
+    },
+    {
+      name: 'hide',
+      enabled: disabled,
+    },
+  ];
+
+  const { styles, attributes, update } = usePopper(
+    referenceElement,
+    popperElement,
+    {
+      placement: anchor,
+      modifiers: popperModifiers,
+      strategy: position,
+    },
+  );
+
+  // only limit tooltip width when the tooltip is a string
+  const maxWidth = typeof tooltip === 'string' ? '200px' : 'fit-content';
+
+  // Manage tooltip attributes
+  const showTooltip = () => popperElement?.setAttribute('data-show', '');
+  const hideTooltip = () => popperElement?.removeAttribute('data-show');
+
+  // Manage tooltip visibility
+  const mouseIn = () => {
+    // popper.js doesn't take into account layout changes, so we need to update it manually
+    update();
+    if (!disabled) showTooltip();
+  };
+  const mouseOut = () => hideTooltip();
+
+  useEffect(() => {
+    if (disabled) hideTooltip();
+  }, [disabled, children, referenceElement]);
+
   return (
-    <span
-      aria-hidden="true"
-      className={classNames(`rc-tooltip-container`, className)}
-      onClick={onClick}
-      onKeyPress={onClick}
-      style={{ ...style }}
-    >
-      {enabled && !!children && !!tooltip && (
-        <div className={classNames('rc-tooltip', `rc-tooltip-${anchor}`)}>
-          {tooltip}
-        </div>
+    <>
+      {!!children && !!tooltip && (
+        <Portal>
+          {/* eslint-disable-next-line */}
+          <div
+            id={tooltipId}
+            className={classNames('rc-tooltip', className)}
+            ref={setPopperElement}
+            style={{ ...styles.popper, textAlign, maxWidth, ...style }}
+            {...attributes.popper}
+            onMouseEnter={mouseIn}
+            onMouseLeave={mouseOut}
+            onClick={onClick}
+            role="tooltip"
+          >
+            {popperOptions.arrow && (
+              <span
+                id="rc-tooltip-arrow"
```

Comment:
Can this be a class instead of an id?

Reply:
No, the arrow for popper doesn't work  without an id